src/libpriv/rpmostree-core.cxx

```diff
@@ -4576,8 +4575,7 @@ rpmostree_context_assemble (RpmOstreeContext      *self,
     }
 
   /* Undo the /etc move above */
-  if (!ror_tempetc_redo_usretc (etc_guard, error))
-    return FALSE;
+  etc_guard->undo();
```

Comment:
I want to emphasize how representative this example will be I think of future code.  Notice that we start using C++ method invocation syntax on a struct, and we use C++ exceptions instead of GError.

Personally I think the first is a notable win (on top of how we didn't need to manually create unsafe C wrappers).  I'm less happy with C+_ exceptions but I think on balance if we can greatly accelerate oxidation it will be worth the short term hit.